### PR TITLE
Restructure recovery steps

### PR DESCRIPTION
**Pull request by tjungblu**
Clarify different aspects of restoring from snapshots, add more descriptive headers and sections to the existing docs.

Note, I will also update this for the 3.5 once we're happy with the structure and content of the recovery doc.

### CODE REVIEW

File: content/en/docs/v3.6/op-guide/recovery.md [hunk 1]
```diff
@@ -18,13 +18,86 @@ Recovering a cluster first needs a snapshot of the keyspace from an etcd member.
...
 
+### Revision Difference
+
+When you are restoring a cluster, existing clients may perceive the revision going back by many hundreds or thousands. This is due to the fact that a given snapshot only contains the data lineage up until the point of when it was taken, whereas the current state might already be further ahead.
+
+This is particularly a problem when running Kubernetes using etcd, where controllers and operators may use so called `informers` which act as local caches and get notified on updates using watches. Restoring to an older revision may not correctly refresh the caches, causing unpredictable and inconsistent behavior in the controllers.
+
+When restoring from a snapshot in the context of either: known consumers of the watch API, local cached copies of etcd data or when using Kubernetes in general - it is highly recommended to restore using "revision bumps" below. 
+
+### Restoring from database file
```
Comment (ahrtr) [inline content/en/docs/v3.6/op-guide/recovery.md:46]: The snapshot can be the db file or a snapshot created by `etcdctl snapshot save` command.

```suggestion
### Restoring from snapshot
```

File: content/en/docs/v3.6/op-guide/recovery.md [hunk 2]
```diff
@@ -64,10 +137,4 @@ $ etcd \
   --listen-peer-urls http://host3:2380 &
 ```
 
-Now the restored etcd cluster should be available and serving the keyspace given by the snapshot.
-
-## Restoring a cluster from membership mis-reconfiguration with wrong URLs
```
Comment (ahrtr) [inline content/en/docs/v3.6/op-guide/recovery.md:69]: Suggest to keep this section. We should clearly call out the workaround `--force-new-cluster` although it isn't recommended.

Comment (tjungblu) [reply content/en/docs/v3.6/op-guide/recovery.md:69]: I have merged the useful parts into `Restoring with updated membership` above, not sure the issue links for 3.2 are very relevant nowadays. Hence the removal.

File: content/en/docs/v3.6/op-guide/recovery.md (hunk 1 above)
Comment (tjungblu) [reply content/en/docs/v3.6/op-guide/recovery.md]: updated

File: content/en/docs/v3.6/op-guide/recovery.md [hunk 3]
```diff
@@ -18,13 +18,86 @@ Recovering a cluster first needs a snapshot of the keyspace from an etcd member.
...
+
+A full invocation may look like this:
+
+```sh
+$ etcdutl snapshot restore snapshot.db --bump-revision 1000000000 --mark-compacted --data-dir output-dir
+```
+
+### Restoring with updated membership
+
+The members of an etcd cluster are stored in etcd itself and maintained through the raft consensus algorithm. When quorum is lost entirely, you may want to reconsider where and how the new cluster is formed, for example, on an entirely new set of servers.
```
Comment (ahrtr) [inline content/en/docs/v3.6/op-guide/recovery.md:75]: nit

```suggestion
The members of an etcd cluster are stored in etcd itself and maintained through the raft consensus algorithm. When quorum is lost entirely, you may want to reconsider where and how the new cluster is formed, for example, on an entirely new set of members.
```

File: content/en/docs/v3.5/op-guide/recovery.md [hunk 4]
```diff
@@ -18,13 +18,86 @@ Recovering a cluster first needs a snapshot of the keyspace from an etcd member.
...
+
+A full invocation may look like this:
+
+```sh
+$ etcdutl snapshot restore snapshot.db --bump-revision 1000000000 --mark-compacted --data-dir output-dir
+```
+
+### Restoring with updated membership
+
+The members of an etcd cluster are stored in etcd itself and maintained through the raft consensus algorithm. When quorum is lost entirely, you may want to reconsider where and how the new cluster is formed, for example, on an entirely new set of servers.
```
Comment (ahrtr) [inline content/en/docs/v3.5/op-guide/recovery.md:75]:
```suggestion
The members of an etcd cluster are stored in etcd itself and maintained through the raft consensus algorithm. When quorum is lost entirely, you may want to reconsider where and how the new cluster is formed, for example, on an entirely new set of members.
```

**Review by ahrtr — APPROVED** [review]
lgtm with very minor comments

Thanks.

File: content/en/docs/v3.6/op-guide/recovery.md (hunk 3 above)
Comment (tjungblu) [reply content/en/docs/v3.6/op-guide/recovery.md]: fixed

File: content/en/docs/v3.5/op-guide/recovery.md (hunk 4 above)
Comment (tjungblu) [reply content/en/docs/v3.5/op-guide/recovery.md]: fixed

**Review by jmhbnz — APPROVED** [review]
LGTM - Many thanks @tjungblu this is a great clarification.

**Pull request merged**